jupyterlab/handlers/yjs_echo_ws.py

```diff
@@ -56,17 +56,23 @@ def on_message(self, message):
         room = cls.rooms.get(room_id)
         if message[0] == ServerMessageType.ACQUIRE_LOCK:
             now = int(time.time())
-            if room.lock is None or now - room.lock > 15: # no lock or timeout
+            if room.lock is None or now - room.lock > (10 * len(room.clients)) : # no lock or timeout
                 room.lock = now
+                room.lock_holder = self
                 # print('Acquired new lock: ', room.lock)
                 # return acquired lock
                 self.write_message(bytes([ServerMessageType.ACQUIRE_LOCK]) + room.lock.to_bytes(4, byteorder = 'little'), binary=True)
+            
+            elif room.lock_holder == self :
```

Comment:
Did you make sure that the current implementation supports renewing their lock?

Reply:
Yes, I test it and looks like it works.